Sources/CardPayments/APIRequests/ConfirmPaymentSourceRequest.swift

```diff
@@ -9,15 +9,19 @@ struct ConfirmPaymentSourceRequest: APIRequest {
     private let orderID: String
     private let pathFormat: String = "/v2/checkout/orders/%@/confirm-payment-source"
     private let accessToken: String
-    private let jsonEncoder = JSONEncoder()
+    var jsonEncoder: JSONEncoder
     
     /// Creates a request to attach a payment source to a specific order.
     /// In order to use this initializer, the `paymentSource` parameter has to
     /// contain the entire dictionary as it exists underneath the `payment_source` key.
+    
     init(
         accessToken: String,
-        cardRequest: CardRequest
+        cardRequest: CardRequest,
+        encoder: JSONEncoder = JSONEncoder()
     ) throws {
+        // encode with custom encoder that throws error if passed in
```

Comment:
```suggestion
```

Don't think this comment is necessary. What would be more appropriate is adding a comment next to the default param like

```
        encoder: JSONEncoder = JSONEncoder() // exposed for test injection
```

Reply:
Addressed here: 7f57a8067d0624219df2ff69d2edf49deaf397c5